README.md

```diff
@@ -1,128 +1,102 @@
-import { wordlist } from "ethereum-cryptography/bip39/wordlists/english.js";
-
-// utilities
-import { hexToBytes, toHex, utf8ToBytes } from "ethereum-cryptography/utils.js";
-```
-
-## Hashes: SHA256, keccak-256, RIPEMD160, BLAKE2b
-```typescript
-function sha256(msg: Uint8Array): Uint8Array;
-function sha512(msg: Uint8Array): Uint8Array;
-function keccak256(msg: Uint8Array): Uint8Array;
-function ripemd160(msg: Uint8Array): Uint8Array;
-function blake2b(msg: Uint8Array, outputLength = 64): Uint8Array;
-```
-
-Exposes following cryptographic hash functions:
-
-- SHA2 (SHA256, SHA512)
-- keccak-256 variant of SHA3 (also `keccak224`, `keccak384`,
-and `keccak512`)
-- RIPEMD160
-- BLAKE2b
+primitive. The reason for this is that importing everything from a single file will lead to huge bundles when using this package for the web. This could be
+avoided through tree-shaking, but the possibility of it not working properly
+on one of [the supported bundlers](#browser-usage) is too high.
+
+* [Usage](#usage)
+  * [Dependencies](#dependencies)
+  * [hashes: sha256, sha512, keccak, ripemd160, blake2b](#hashes-sha256-sha512-keccak-ripemd160-blake2b)
+  * [kdfs: pbkdf2, scrypt](#kdfs-pbkdf2-scrypt)
+  * [random: secure randomness](#random-secure-randomness)
+  * [secp256k1: curve operations](#secp256k1-curve-operations)
+  * [bn: pairing-friendly curve](#bn-pairing-friendly-curve)
+  * [bls: pairing-friendly curve](#bls-pairing-friendly-curve)
+  * [aes: encryption](#aes-encryption)
+  * [hdkey: bip32 HD wallets](#hdkey-bip32-hd-wallets)
+  * [bip39: mnemonic phrases](#bip39-mnemonic-phrases)
+  * [math: utilities](#math-utilities)
+  * [utils: generic utilities](#utils-generic-utilities)
+  * [secp256k1-compat: compatibility layer with other libraries](#secp256k1-compat-compatibility-layer-with-other-libraries)
+  * [All imports](#all-imports)
+* [Caveats](#caveats)
+  * [AES](#aes)
+      * [Encrypting with passwords](#encrypting-with-passwords)
+      * [Operation modes](#operation-modes)
+      * [Padding plaintext messages](#padding-plaintext-messages)
+      * [How to use the IV parameter](#how-to-use-the-iv-parameter)
+      * [How to handle errors with this module](#how-to-handle-errors-with-this-module)
+* [Browser usage](#browser-usage)
+  * [Rollup setup](#rollup-setup)
+* [Upgrading](#upgrading)
+  * [Changelog](#changelog)
+  * [From v2 to v3](#from-v2-to-v3)
+  * [From v1 to v2](#from-v1-to-v2)
+  * [From v0.1 to v1](#from-v01-to-v1)
+* [Security](#security)
+* [License](#license)
+
+### Dependencies
+
+All functionality of the module is simple
+re-export of 6 audited [noble & scure libraries](https://paulmillr.com/noble/):
+
+- noble-curves, noble-ciphers, noble-hashes
+- scure-base, scure-bip32, scure-bip39
+
+ethereum-cryptography pins versions of the libraries to ensure good
+protection against supply chain attacks. Ideally, your app would also
+pin version of ethereum-cryptography. That means, no `^3.0.0` - use `3.0.0` instead.
```

Comment:
```suggestion
ethereum-cryptography pins versions of the libraries to ensure good
protection against supply chain attacks.
```

Nit: It's a fair strategy for `ethereum-cryptography` why opine on general best practices here?

Reply:
Because it's cryptography / important dep. The suggestion matters less for other deps.

If users pin version of e-c, that means less chance of bad stuff happening. Attacker with stolen NPM key can publish garbage in the package, replacing all noble implementations with malware.